lib/iris/_representation.py

```diff
@@ -66,7 +66,11 @@ def _summary_coord_extra(self, cube, coord):
                         vary.add(key)
                         break
                     value = similar_coord.attributes[key]
-                    if attributes.setdefault(key, value) != value:
+                    # Like "if attributes.setdefault(key, value) != value:"
+                    # ..except setdefault fails if values are numpy arrays.
+                    if key not in attributes:
+                        attributes[key] = value
+                    elif not np.all(attributes[key] == value):
```

Comment:
It looks like there exists logic in the metadata to handle this kind of comparison: https://github.com/SciTools/iris/blob/a136915a8421864a4bb18dcc06bade3b324ff1f3/lib/iris/common/metadata.py#L51
This would handle edge cases differently (dtype, masked arrays etc.)

Reply:
Yes, that really does look like the right way.